sql/src/main/java/org/apache/apex/malhar/sql/table/KafkaEndpoint.java

```diff
@@ -0,0 +1,134 @@
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.apex.malhar.sql.table;
+
+import java.util.Map;
+import java.util.Properties;
+
+import org.apache.apex.malhar.kafka.KafkaSinglePortInputOperator;
+import org.apache.apex.malhar.kafka.KafkaSinglePortOutputOperator;
+import org.apache.apex.malhar.sql.operators.OperatorUtils;
+import org.apache.apex.malhar.sql.planner.RelInfo;
+import org.apache.calcite.adapter.java.JavaTypeFactory;
+import org.apache.calcite.rel.type.RelDataType;
+import org.apache.calcite.rel.type.RelDataTypeFactory;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.producer.ProducerConfig;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Lists;
+
+import com.datatorrent.api.DAG;
+import com.datatorrent.api.Operator;
+
+/**
+ * This is an implementation of {@link Endpoint} which defined how data should be read/written from kafka messaging system
+ */
+public class KafkaEndpoint implements Endpoint
+{
+  public static final String KEY_DESERIALIZER = "org.apache.kafka.common.serialization.StringDeserializer";
+  public static final String VALUE_DESERIALIZER = "org.apache.kafka.common.serialization.StringDeserializer";
+  public static final String KEY_SERIALIZER = "org.apache.kafka.common.serialization.StringSerializer";
+  public static final String VALUE_SERIALIZER = "org.apache.kafka.common.serialization.StringSerializer";
+
+
+  public static final String KAFKA_SERVERS = "servers";
+  public static final String KAFKA_TOPICS = "topics";
+
+  private MessageFormat messageFormat;
+
+  private Map<String, Object> operands;
+
+  public KafkaEndpoint()
+  {
+  }
+
+  public KafkaEndpoint(String kafkaServers, String topics, MessageFormat messageFormat)
+  {
+    this.messageFormat = messageFormat;
+    this.operands = ImmutableMap.<String, Object>of(KAFKA_SERVERS, kafkaServers, KAFKA_TOPICS, topics);
+  }
+
+  @Override
+  public EndpointType getTargetType()
+  {
+    return EndpointType.KAFKA;
+  }
+
+  @Override
+  public void setEndpointOperands(Map<String, Object> operands)
+  {
+    this.operands = operands;
+  }
+
+  @Override
+  public void setMessageFormat(MessageFormat messageFormat)
+  {
+    this.messageFormat = messageFormat;
+  }
+
+  @Override
+  public RelInfo populateInputDAG(DAG dag, JavaTypeFactory typeFactory)
+  {
+    KafkaSinglePortInputOperator kafkaInput = dag.addOperator(OperatorUtils.getUniqueOperatorName("KafkaInput"),
+        KafkaSinglePortInputOperator.class);
+    kafkaInput.setTopics((String)operands.get(KAFKA_TOPICS));
```

Comment:
https://issues.apache.org/jira/browse/APEXMALHAR-2311 as a follow up task should take care of this.